server/src/services/library.service.spec.ts

```diff
@@ -247,49 +259,85 @@ describe(LibraryService.name, () => {
         exclusionPatterns: [],
       });
     });
+  });
 
+  describe('handleOfflineCheck', () => {
     it('should set missing assets offline', async () => {
-      const mockLibraryJob: ILibraryRefreshJob = {
-        id: libraryStub.externalLibrary1.id,
-        refreshModifiedFiles: false,
-        refreshAllFiles: false,
+      const mockAssetJob: ILibraryOfflineJob = {
+        id: assetStub.external.id,
+        importPaths: ['/'],
       };
 
-      libraryMock.get.mockResolvedValue(libraryStub.externalLibrary1);
-      assetMock.getLibraryAssetPaths.mockResolvedValue({
-        items: [assetStub.image],
-        hasNextPage: false,
-      });
+      assetMock.getById.mockResolvedValue(assetStub.external);
 
-      await sut.handleQueueAssetRefresh(mockLibraryJob);
+      storageMock.checkFileExists.mockResolvedValue(false);
 
-      expect(assetMock.updateAll).toHaveBeenCalledWith([assetStub.image.id], { isOffline: true });
-      expect(assetMock.updateAll).not.toHaveBeenCalledWith(expect.anything(), { isOffline: false });
-      expect(jobMock.queueAll).not.toHaveBeenCalled();
+      await sut.handleOfflineCheck(mockAssetJob);
+
+      expect(assetMock.update).toHaveBeenCalledWith({ id: assetStub.external.id, isOffline: true });
```

Comment:
I think it would be nice to also expect the response here.

Reply:
Done